### PR TITLE
adding matrix notifier

### PR DESCRIPTION
**Pull request by antibios**
### Motivation for changes:
Matrix is a secure msging platform that allows for updates via a REST call

### Detailed changes:
Addition of matrix.py to the notifiers folder. 

### Addressed issues:
- Fixes # .
- Feature request.

### Implemented feature requests:
- Matrix support

### Config usage if relevant (new plugin or updated schema):
```
Sample config:
      entries:
        via:
          - matrix:
                  server: 'https://matrix.org'
                  token: 'SECRETKEY-dsarasfaFASZgo'
                  roomid: '!ufPOdHzBzMRGDSASgf:matrix.org'

```
### Log and/or tests output (preferably both):
```
2022-08-23 14:32:49 VERBOSE  notify        sort-stuff     Successfully sent a notification to `matrix`
```
#### To Do:
- [ ] Group multiple notifications to a single msg to reduce number of requests.

### CODE REVIEW

File: flexget/components/notify/notifiers/matrix.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+class MatrixNotifier:
+    """
+    Sends messages via Matrix. The MatrixHttpApi library is required to be installed.
+    Install it with: `pip install matrix_client`
+
+    All fields are required.
+
+    token: Is available in Element under Help/About.
+    roomId: Is available in Element under room settings.
```
Comment (gazpachoking) [inline flexget/components/notify/notifiers/matrix.py:33]: Capitalization is different here than in actual usage. Maybe we want to standardize it to `room_id` everywhere?

**Review by gazpachoking — COMMENTED** [review]
Looks pretty simple and good. Maybe tweak the room id name in config?

File: flexget/components/notify/notifiers/matrix.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    def notify(self, title, message, config):
+        """
+        Send notification to Matrix Room
+        """
+        notification = {'body': message, 'msgtype': "m.text"}
+        room = urljoin(
+            config['server'],
+            "_matrix/client/r0/rooms",
+            config['roomid'],
+            "send/m.room.message?access_token=" + config['token'],
```
Comment (paranoidi) [inline flexget/components/notify/notifiers/matrix.py:68]: probably should use

urllib.parse.urlencode(....)

File: flexget/components/notify/notifiers/matrix.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+from flexget.utils.requests import Session as RequestSession
+
+requests = RequestSession(max_retries=3)
+
+plugin_name = 'matrix'
+
+logger = logger.bind(name=plugin_name)
+
+
+def urljoin(*args):
```
Comment (paranoidi) [inline flexget/components/notify/notifiers/matrix.py:16]: why not just use standard urljoin ?

Comment (antibios) [reply flexget/components/notify/notifiers/matrix.py:16]: urllib.urljoin() takes from 2 to 3 positional arguments but 4 were required.

File: flexget/components/notify/notifiers/matrix.py (hunk 1 above)
Comment (antibios) [reply flexget/components/notify/notifiers/matrix.py:33]: done

File: flexget/components/notify/notifiers/matrix.py (hunk 2 above)
Comment (antibios) [reply flexget/components/notify/notifiers/matrix.py:68]: rllib.urljoin() takes from 2 to 3 positional arguments but 4 were required.

**Comment by paranoidi** [comment]
Awesome, let's merge this in. Can you update the wiki too?

I created the page at

https://flexget.com/en/Plugins/Notifiers/matrix

In case I have completed it, just proof-read it :)

**Comment by antibios** [comment]
I've updated the wiki with instructions

**Comment by paranoidi** [comment]
@antibios nice! I made some further formatting improvements to stay more consistent with other documentation

thanks for contributing

**Comment by gyandoo** [comment]
@paranoidi is it possible to send text/markdown as matrix supports it, and if yes, how, thanks

**Pull request merged**